packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -1752,6 +1754,73 @@ The Flutter Preview device does not support the following plugins from your pubs
       );
     });
   });
+
+  testUsingContext('exits tool when deleting .plugin_symlinks fails', () async {
```

Comment:
I'm somewhat concerned about having tests like these. The existence of this test implies that every single file operation throughout the entire tool should have one or more tests for possibility of the operation failing.  Do tests like these add more value than they produce noise?

Said more simply, I only wrote this test to satisfy our testing requirement, not because I think it's a good test to have.

Reply:
> The existence of this test implies that every single file operation throughout the entire tool should have one or more tests for possibility of the operation failing. Do tests like these add more value than they produce noise?

I think there's a difference between testing every single operation to verify it behaves as expected and adding a test for a specific operation that was known to be problematic to verify someone doesn't undo it.

That having been said, I don't care that much, and your cache_test update would have made the bot happy.

Reply:
> I think there's a difference between testing every single operation to verify it behaves as expected and adding a test for a specific operation that was known to be problematic to verify someone doesn't undo it.

I suppose.

🤷 I do think the test was worth writing just to verify the change worked, and I therefore think it's useful for future archeologists to verify that the change was tested before being landed. So I think I'll keep writing these.